MeesGame/Gameplay/GameObjects/EndTile.cs

```diff
@@ -0,0 +1,27 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace MeesGame.Gameplay.GameObjects
+{
+    class EndTile : FloorTile
+    {
+        GameStateManager gameStates;
+
+        public EndTile(int layer = 0, string id = "") : base("end_door", TileType.Floor, layer, id)
+        {
+
+        }
+
+        //Check if the player finished the level
+        public void Finished(Player player)
+        {
+            if (player.Position == this.position)
+                gameStates.SwitchTo("GameOverState");
```

Comment:
gameStates is null at this point. Change it to GameEnvironment.GameStateManager.SwitchTo (or similar)